Microsoft.Research/Contracts/WindowsBase/System.Windows.Threading.DispatcherObject.cs

```diff
@@ -22,7 +22,26 @@ public abstract class DispatcherObject
   {
     protected DispatcherObject() { }
 
-    public Dispatcher Dispatcher { get { Contract.Ensures(Contract.Result<Dispatcher>() != null); return null; } }
+    // DetachFromDispatcher is e.g. called by Freezable.Freeze()
+    //
+    // This method allows certain derived classes to break the dispatcher affinity
+    // of our objects.
+    // [FriendAccessAllowed] // Built into Base, also used by Framework.
+    internal void DetachFromDispatcher()
+    {
+      // _dispatcher = null;
+    }
+
+    public Dispatcher Dispatcher
+    {
+      get
+      {
+        // Note: a DispatcherObject that is not associated with a
```

Comment:
:bulb: Looking at the source code for this class, I can confirm there are cases where this property returns `null`. However, it seems the particular case that users will recognize is `Freezable` objects which have been frozen. I recommend updating the comment to clarify this case:

> Freezable instances which have been frozen are detached from the Dispatcher, so this property will return null.